hnn_core/cell.py

```diff
@@ -783,9 +793,22 @@ def record(self, record_vsec=False, record_isec=False):
 
                 for syn_name in self.isec[sec_name]:
                     self.isec[sec_name][syn_name] = h.Vector()
-
                     self.isec[sec_name][syn_name].record(
                         self._nrn_synapses[syn_name]._ref_i)
+        
+        # calcium concentration
+        if record_ca == 'soma':
+            self.ca = dict.fromkeys(['soma'])
+        elif record_ca == 'all':
+            self.ca = dict.fromkeys(section_names)
+
+        if record_ca:     
+            for sec_name in self.ca:
+                if 'ca' in self.sections[sec_name].mechs:
```

Comment:
```suggestion
                if hasattr(self._nrn_sections[sec_name](0.5), '_ref_cai)::
```
Your suggestion! Just making sure we don't forget